api/python/t4/packages.py

```diff
@@ -218,9 +218,8 @@ def fetch(self, dest=None):
             None
         """
         if dest is None:
-            name = pathlib.Path(_to_singleton(self.physical_keys)).name
-            name = name.split('?versionId=')[0]
-            dest = name
+            pkey = _to_singleton(self.physical_keys)
+            dest = pathlib.PurePosixPath(unquote(urlparse(pkey).path)).name
```

Comment:
You should then make a proper URL out of this, and move `fix_url` below into an `else:`. E.g.:
```
name = pathlib.PurePosixPath(unquote(urlparse(pkey).path)).name
dest = (pathlib.Path().resolve() / name).as_uri()
```
(If you're downloading e.g. `s3://bucket/~user`, you probably don't want to expand the `~`.)

Reply:
(That said, `name` could end up being `..` - that's also bad.)

Reply:
Name would only end up being `..` if `pathlib.PurePosixPath([...])` returns e.g. `a/b/..`, which I guess is possible but it would be _really weird_ if someone named their file `..`. 

You are right about not needing or wanting to run `fix_url` on this logic's output, I see how this would cause problems with names starting in `~` or in `..`. I moved it under an `else`.